### ISSUE TITLE
Consider all number as float/double for a better user experience

### ISSUE THREAD

**Issue by MrGorki** [davideicardi/DynamicExpresso#35]
Hello,

I am giving "440*(85/440)" to interpreter but it returns 0. Do anyone have any idea about why?

**Comment by davideicardi** [comment]
This is the standard C# behavior.
Basically 85 and 440 are considered integer and so 85/440 return just 0.

You should write: "440*(85.0/440.0)" so that the interpreter consider it as double.

**Comment by MrGorki** [comment]
Thanks but I should convert all the integers to double in this way. Isn't it better to be handled by the library?

**Comment by davideicardi** [comment]
Yes, I understand your point. Unfortunately DynamicExpresso is more like a C# interpreter, not strictly a math interpreter. Anyway maybe this can be a future enhancement. 

Of course any help is appreciate :-) ...

**Comment by davideicardi** [comment]
For now as a workaround you can probably use a regular expression before calling the interpreter that replace integers with double. This should be quite easy I think...

**Comment by MrGorki** [comment]
Great! Thank you! I will try to investigate it on library itself also! Cheers!

**Comment by eruspravus** [comment]
A quick dirty fix to the problem requires some changes to Parsing/Parser.cs;

```
		Expression ParseIntegerLiteral()
		{
			ValidateToken(TokenId.IntegerLiteral);
			string text = _token.text;
			if (text[0] != '-')
			{
				ulong value;
				if (!UInt64.TryParse(text, ParseLiteralUnsignedNumberStyle, ParseCulture, out value))
					throw CreateParseException(_token.pos, ErrorMessages.InvalidIntegerLiteral, text);

				NextToken();

				if (value <= (ulong)Int32.MaxValue) 
					return CreateLiteral((double)value, text);
				if (value <= (ulong)UInt32.MaxValue) 
					return CreateLiteral((double)value, text);
				if (value <= (ulong)Int64.MaxValue) 
					return CreateLiteral((double)value, text);

				return CreateLiteral(value, text);
			}
			else
			{
				long value;
				if (!Int64.TryParse(text, ParseLiteralNumberStyle, ParseCulture, out value))
					throw CreateParseException(_token.pos, ErrorMessages.InvalidIntegerLiteral, text);

				NextToken();

				if (value >= Int32.MinValue && value <= Int32.MaxValue)
					return CreateLiteral((double)value, text);

				return CreateLiteral(value, text);
			}
		}
```
I'm still a bit unsure what other effects this could have to the expressions.

But this works;
![parsingfixed](https://cloud.githubusercontent.com/assets/5288253/10728914/2107a69c-7be6-11e5-87e2-f9e827f333d6.PNG)


If I did something nicer (maybe a bool ForceNumericToDouble property), would you be interested in a commit for this?

**Comment by davideicardi** [comment]
Thank you @eruspravus .
I think that problems can occur when you use the numbers for something else instead of only math expressions, for example if you call functions that expect an integer or something similar. 

Maybe we can activate this feature using a new value of the `InterpreterOptions` enum. What do you think? (this is a enum flags used already for other similar configuration).

Finally consider that now Roslyn scripting is out](https://joshvarty.wordpress.com/2015/10/15/learn-roslyn-now-part-14-intro-to-the-scripting-api/), so I'm not sure if DynamicExpresso can still have some value... Roslyn is far more powerful and probably it is the way to go. I need to study it better.

**Comment by davideicardi** [comment]
Closed by #146
